### PR TITLE
feat: Support Pageable, Page and UUID in endpoints

### PR DESCRIPTION
**Pull request by Artur-**
Fixes https://github.com/vaadin/flow/issues/10207
Fixes https://github.com/vaadin/fusion/issues/18
Fixes https://github.com/vaadin/fusion/issues/17

### CODE REVIEW

File: fusion-endpoint/src/test/java/com/vaadin/fusion/endpointransfermapper/EndpointTransferMapperTest.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+public class EndpointTransferMapperTest {
+
+    private EndpointTransferMapper endpointTransferMapper = new EndpointTransferMapper();
+
+    @Test
+    public void getTransferTypeClass_works() {
+        Assert.assertEquals(PageableDTO.class,
+                endpointTransferMapper.getTransferType(Pageable.class));
+        // Assert.assertEquals(PageableDTO.class,
```
Comment (haijian-vaadin) [inline fusion-endpoint/src/test/java/com/vaadin/fusion/endpointransfermapper/EndpointTransferMapperTest.java:24]: should these be removed or uncommented?

File: fusion-endpoint/src/test/java/com/vaadin/fusion/generator/endpoints/AbstractEndpointGenerationTest.java [hunk 2]
```diff
@@ -157,6 +161,20 @@ private void verifyOpenApiObject() {
             Map<String, Schema> componentSchemas = Optional
                     .ofNullable(actualOpenAPI.getComponents())
                     .map(Components::getSchemas).orElse(Collections.emptyMap());
+
+            componentSchemas.keySet().removeIf(clsName -> {
+                /* Skip classes that are added because of the mappers */
+                if (clsName.startsWith(
+                        PageableDTO.class.getPackage().getName() + ".")) {
+                    return true;
+                }
```
Comment (haijian-vaadin) [inline fusion-endpoint/src/test/java/com/vaadin/fusion/generator/endpoints/AbstractEndpointGenerationTest.java:170]: maybe better to extract to a separate function?

**Review by haijian-vaadin — CHANGES_REQUESTED** [review]

File: fusion-endpoint/src/test/java/com/vaadin/fusion/generator/endpoints/AbstractEndpointGenerationTest.java (hunk 2 above)
Comment (Artur-) [reply fusion-endpoint/src/test/java/com/vaadin/fusion/generator/endpoints/AbstractEndpointGenerationTest.java:170]: Done

File: fusion-endpoint/src/test/java/com/vaadin/fusion/endpointransfermapper/EndpointTransferMapperTest.java (hunk 1 above)
Comment (Artur-) [reply fusion-endpoint/src/test/java/com/vaadin/fusion/endpointransfermapper/EndpointTransferMapperTest.java:24]: Uncommented, done

**Review by haijian-vaadin — APPROVED** [review]

**Pull request merged**